### PR TITLE
feat: Trade Market Hide Prompt option [skip ci]

### PR DESCRIPTION
**Pull request by aprevrha**
parity for #2908

### CODE REVIEW

**Comment by aprevrha** [comment]
Just noticed this clashes with quick search (#2908)

**Comment by aprevrha** [comment]
fixed with [5a269d6](https://github.com/Wynntils/Wynntils/pull/2908/commits/5a269d6e0f7d1c75f3782f630b6fe48dde96f956)

File: common/src/main/java/com/wynntils/features/trademarket/TradeMarketAutoOpenChatFeature.java [hunk 1]
```diff
@@ -34,6 +40,9 @@ public void onChatMessageReceive(ChatMessageReceivedEvent event) {
 
         if (styledText.matches(TYPE_TO_CHAT_PATTERN)) {
             openChatWhenContainerClosed = true;
+            if (hidePrompt.get()) {
+                event.setCanceled(true);
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/features/trademarket/TradeMarketAutoOpenChatFeature.java:51]: As other features rely on this chat message, you should set the event priority to lower so this doesn't cancel it before they use it

**Comment by magicus** [comment]
With all these trademarket changes going on, I believe the time has come to create a `TrademarketModel`. There is too much knowledge about the trademarket spread over too many features, and now they start to step on each other's toes.

A TrademarketModel could have knowledge about what state the TM is in (as in which screen is currently shown), perform searches on request, etc.

**Comment by ShadowCat117** [comment]
There already is a `TradeMarketModel` but I believe it's mainly used for the custom tm screen feature which has yet to be updated for 2.1

File: common/src/main/java/com/wynntils/features/trademarket/TradeMarketAutoOpenChatFeature.java (hunk 1 above)
Comment (antonprev) [reply common/src/main/java/com/wynntils/features/trademarket/TradeMarketAutoOpenChatFeature.java:51]: Done, set it to LOW

**Comment by aprevrha** [comment]
> A TrademarketModel could have knowledge about what state the TM is in (as in which screen is currently shown), perform searches on request, etc.

> There already is a `TradeMarketModel` but I believe it's mainly used for the custom tm screen feature which has yet to be updated for 2.1

Agreed, the states should all be in the model. Updating the custom tm screen would be the perfect opportunity to do so. 
I haven't worked with models yet, but could get started, will need some help tho :)

For now I'd like to merge this anyways as the issue should be temporarily fixed

**Comment by ShadowCat117** [comment]
> I haven't worked with models yet, but could get started, will need some help tho :)

They're fairly straightforward, essentially it's just collecting information about a certain Wynncraft related system in one place that other components can access. Raid model for example does everything to do with tracking when you are in a raid, what room you are in, how long you've been in that room for etc. Then the functions just query the model for that information instead of them handling it themselves.

As always if you need any assistance the discord channel is there and someone will be around

**Review by ShadowCat117 — APPROVED** [review]
Thanks!

**Pull request merged**